fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java

```diff
@@ -256,24 +257,32 @@ public CostEstimate visitPhysicalDistribution(PhysicalDistributionOperator node,
             Preconditions.checkNotNull(statistics);
 
             CostEstimate result;
+            SessionVariable sessionVariable = ConnectContext.get().getSessionVariable();
             DistributionSpec distributionSpec = node.getDistributionSpec();
             switch (distributionSpec.getType()) {
                 case ANY:
                     result = CostEstimate.ofCpu(statistics.getOutputSize(outputColumns));
                     break;
                 case BROADCAST:
-                    if (statistics.getOutputSize(outputColumns) >
-                            ConnectContext.get().getSessionVariable().getMaxExecMemByte()) {
+                    if (statistics.getOutputSize(outputColumns) > sessionVariable.getMaxExecMemByte()) {
                         return CostEstimate.infinite();
                     }
                     int parallelExecInstanceNum = Math.max(1, getParallelExecInstanceNum(context));
+                    int pipelineDop = 1;
+                    if (sessionVariable.isEnablePipelineEngine()) {
+                        // pipelineDop is 0 means that the query executed in pipeline engine use the half number of cpu
+                        // cores, we use 32 as an estimate value
+                        pipelineDop = sessionVariable.getPipelineDop() > 0 ? sessionVariable.getPipelineDop() : 32;
```

Comment:
Add a TODO here.

32 isn't reasonable here.  `half number of cpu cores` could be any value. we should get this information from BE heartbeat.

Reply:
done